apt-bootstrap

```diff
@@ -342,7 +327,18 @@ class AptBootstrap(object):
             if not os.path.exists(path):
                 logging.info('Creating dev symlink %s' % path)
                 os.symlink(link[1], path)
-        
+
+        # Inside a container, we might not be allowed to create /dev/ptmx.
+        # If not, do the next best thing.
+        path = os.path.join(self.path, 'dev', 'ptmx')
+        try:
+            os.mknod(path, stat.S_IFCHR, os.makedev(5, 2))
+        except OSError:
+            logging.warning("Could not create /dev/ptmx, falling back to " \
+                            "symlink. This chroot will require /dev/pts " \
+                            "mounted with ptmxmode=666")
+            os.symlink('pts/ptmx', path)
```

Comment:
Thanks. Upstream debootstrap did overhaul this area quite a bit from when I wrote this.

I don't think we need to handle this now (maybe not ever), but https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=872577 is the issue with hit in OBS where debootstrap can't handle an existing /dev. We probably wouldn't ever need that in the ostree builder, but it might be nice to be consistent.